src/main/java/com/azure/cosmos/examples/documentcrud/async/DocumentCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,563 @@
+                .onErrorResume((exception) -> {
+                    logger.error(
+                            "Exception. e: {}",
+                            exception.getLocalizedMessage(),
+                            exception);
+                    return Mono.empty();
+                }).subscribe(listJsonNode -> {
+                    //pass the List<JsonNode> to the instance variable
+                    this.jsonList = listJsonNode;
+                });
+
+    }
+
+    private void replaceDocument() throws Exception {
+        logger.info("Replace document {}", documentId);
+
+        // Replace existing document with new modified document
+        Family family = new Family();
+        family.setLastName(documentLastName);
+        family.setId(documentId);
+        family.setDistrict("Columbia"); // Document modification
+
+        container.replaceItem(family, family.getId(),
+                        new PartitionKey(family.getLastName()), new CosmosItemRequestOptions())
+                .doOnSuccess(itemResponse -> {
+                    logger.info("Request charge of replace operation: {} RU", itemResponse.getRequestCharge());
+                    //set flag so calling method can know when done
+                })
+                .doOnError((exception) -> {
+                    logger.error(
+                            "Exception. e: {}",
+                            exception.getLocalizedMessage(),
+                            exception);
+                }).subscribe();
+        logger.info("replace subscribed: will process asynchronously....");
+    }
+
+    private void upsertDocument() throws Exception {
+        logger.info("Replace document {}", documentId);
+
+        // Replace existing document with new modified document (contingent on
+        // modification).
+        Family family = new Family();
+        family.setLastName(documentLastName);
+        family.setId(documentId);
+        family.setDistrict("Columbia"); // Document modification
+
+        container.upsertItem(family, new CosmosItemRequestOptions())
+                .doOnSuccess(itemResponse -> {
+                    logger.info("Request charge of upsert operation: {} RU", itemResponse.getRequestCharge());
+                    // set flag so calling method can know when done
+                })
+                .doOnError((exception) -> {
+                    logger.error(
+                            "Exception. e: {}",
+                            exception.getLocalizedMessage(),
+                            exception);
+                }).subscribe();
+        logger.info("upsert subscribed: will process asynchronously....");
+    }
+
+    private void replaceDocumentWithConditionalEtagCheck() throws Exception {
+        logger.info("Replace document {}, employing optimistic concurrency using ETag.", documentId);
+
+        // Obtained current document ETag
+        container.readItem(documentId, new PartitionKey(documentLastName), Family.class)
```

Comment:
we can also choose to chain them together instead of using Thread.sleep:
` ` ` 
container.readItem(documentId, new PartitionKey(documentLastName), Family.class)
    .flatmap(response -> {
         Family family = this.family;
        family.setRegistered(!family.isRegistered());
        return container.replaceItem(...)
})
` ` `

Reply:
Thanks - I like this example so I applied this to both etag checking examples (see file).